admin/class-admin-editor-specific-replace-vars.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Determines the editor specific replacement variables.
+ */
+class WPSEO_Admin_Editor_Specific_Replace_Vars {
+
+	/**
+	 * @var array The editor specific replacement variables.
+	 */
+	protected $editor_specific_replace_vars = array(
+		// Posts types.
+		'page'                    => array( 'id', 'pt_single', 'pt_plural', 'parent_title' ),
+		'post'                    => array( 'id', 'term404', 'pt_single', 'pt_plural' ),
+		// Custom post type.
+		'custom_post_type'        => array( 'id', 'term404', 'pt_single', 'pt_plural' ),
+
+		// Taxonomies.
+		'category'                => array( 'term_title', 'term_description', 'category_description', 'parent_title' ),
+		'post_tag'                => array( 'term_title', 'term_description', 'tag_description' ),
+		// Custom taxonomy.
+		'term-in-custom-taxomomy' => array( 'term_title', 'term_description', 'parent_title' ),
+
+		// Settings - archive pages.
+		'custom-post-type_archive' => array(),
+		// Settings - special pages.
+		'search'                  => array( 'searchphrase' ),
+	);
+
+	/**
+	 * WPSEO_Admin_Editor_Specific_Replace_Vars constructor.
+	 */
+	public function __construct() {
+		$this->apply_custom_fields();
+		$this->apply_custom_taxonomies();
+	}
+
+	/**
+	 * Retrieves the shared replacement variable names.
+	 *
+	 * Which are the replacement variables without the editor specific ones.
+	 *
+	 * @param array $replace_vars_list                 The replace vars list.
+	 * @param array $editor_specific_replace_vars_list The editor specific replace var names.
+	 *
+	 * @return array The shared replacement variable names.
+	 */
+	public static function get_shared_replace_vars( $replace_vars_list, $editor_specific_replace_vars_list ) {
+		$filter_values       = self::array_flatten( $editor_specific_replace_vars_list );
+		$shared_replace_vars = array();
+
+		foreach ( $replace_vars_list as $replace_var ) {
+			$name = $replace_var['name'];
+			if ( ! in_array( $name, $filter_values ) ) {
+				$shared_replace_vars[] = $name;
+			}
+		}
+
+		return $shared_replace_vars;
+	}
+
+	/**
+	 * Determines the page type of the current term.
+	 *
+	 * @param string $taxonomy The taxonomy name.
+	 *
+	 * @return string The page type.
+	 */
+	public function determine_for_term( $taxonomy ) {
+		$editor_specific_replace_vars = $this->get_editor_specific_replace_vars();
+		if ( array_key_exists( $taxonomy, $editor_specific_replace_vars ) ) {
+			return $taxonomy;
+		}
+
+		return 'term-in-custom-taxomomy';
+	}
+
+	/**
+	 * Determines the page type of the current post.
+	 *
+	 * @param WP_Post $post The WordPress global post object.
```

Comment:
Although seems to me this method is not actually used, for analogy with the one in `WPSEO_Admin_Recommended_Replace_Vars` seems to me this param is not always necessarily the global post, and it could be a retrieved post instance?

Reply:
This will be needed when implementing for posts and pages. And true, good point, will adapt.